modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[UI]AllowUsersToDefineAPIApplicationEndpoints/AllowUsersToDefineEndpointsForObjectsWithSiteScope.testcase

```diff
@@ -0,0 +1,221 @@
+				value = "c/my-app");
+		}
+	}
+
+	@priority = 4
+	test CanSeePublishedAPIApplicationWithSiteAndCompanyScopedEndpointsWithoutSchemas {
+		property portal.acceptance = "true";
+		property test.liferay.virtual.instance = "false";
+		property test.run.type = "single";
+
+		task ("Given with postAPIApplication to create a published api Application with site scope endpoint and company scope endpoint") {
+			var response = ApplicationAPI.createAPIApplication(
+				baseURL = "my-app",
+				endpointName = "testEndpoint",
+				endpointPath = "/testEndpoint",
+				relatedEndpoint = "true",
+				scopeKey = "group",
+				status = "published",
+				title = "My-app");
+
+			var apiApplicationId = JSONPathUtil.getIdValue(response = ${response});
+
+			EndpointAPI.createAPIEndpoint(
+				apiApplicationId = ${apiApplicationId},
+				name = "myEndpoint",
+				path = "/myEndpoint");
+		}
+
+		task ("When I navigate to /o/c/{apiApplication}") {
+			APIExplorer.navigateToOpenAPI(customObjectPlural = "my-app");
+		}
+
+		task ("Then getScopes{siteScopeEndpointPath}Page is present under default") {
+			AssertTextEquals.assertPartialText(
+				locator1 = "OpenAPI#API_METHOD",
+				method = "getScopeScopeKeyTestEndpointPage",
+				service = "default",
+				value1 = "/testEndpoint");
+		}
+
+		task ("And Then get{companyScopeEndpointPath}Page is present under default") {
+			AssertTextEquals.assertPartialText(
+				locator1 = "OpenAPI#API_METHOD",
+				method = "getMyEndpointPage",
+				service = "default",
+				value1 = "/myEndpoint");
+		}
+	}
+
+	@priority = 4
+	test CanSeePublishedAPIApplicationWithSiteAndCompanyScopedEndpointsWithSchemas {
+		property portal.acceptance = "true";
+		property test.liferay.virtual.instance = "false";
+		property test.run.type = "single";
+
+		task ("Given created an published api application with site scope endpoint and company scope endpoint with related request and response schemas") {
+			var responseFromApplication = ApplicationAPI.createAPIApplication(
+				baseURL = "my-app",
+				endpointName = "testEndpoint",
+				endpointPath = "/testEndpoint",
+				mainObjectDefinitionErc = "L_API_APPLICATION",
+				relatedEndpoint = "true",
+				relatedSchema = "true",
+				schemaName = "testSchema",
+				scopeKey = "group",
+				status = "published",
+				title = "My-app");
+
+			var endpointId1 = JSONUtil.getWithJSONPath(${responseFromApplication}, "$.apiApplicationToAPIEndpoints[*].id");
+			var schemaId1 = JSONUtil.getWithJSONPath(${responseFromApplication}, "$.apiApplicationToAPISchemas[*].id");
+
+			SchemaAPI.relateRequestSchemaToEndpointByIds(
+				endpointId = ${endpointId1},
+				requestSchemaId = ${schemaId1});
+
+			SchemaAPI.relateResponseSchemaToEndpointByIds(
+				endpointId = ${endpointId1},
+				responseSchemaId = ${schemaId1});
+
+			var apiApplicationId = JSONPathUtil.getIdValue(response = ${responseFromApplication});
+
+			var responseFromEndpoint = EndpointAPI.createAPIEndpoint(
```

Comment:
We can check how to improve to create application with multiple endpoints in the future, we're good for now, thanks!